File: LeetCode/Easy/0013-roman-to-integer/0013-roman-to-integer.cs
public class Solution {
    public int RomanToInt(string s) {
        var lastValue = 0;
        var result = 0;
        for (var ii = s.Length - 1; ii >= 0; --ii)
        {
            var currentValue = GetValue(s[ii]);
            if (currentValue >= lastValue)
            {
                result += currentValue;
            }
            else
            {
                result -= currentValue;
            }
            lastValue = currentValue;
        }
        return result;
    }


    private int GetValue(char c) {
        return c switch {
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000,
            _ => 0 // Fallback, choć w zadaniu zakładamy poprawny input
        };
    }
}

File: LeetCode/Easy/0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        mapping = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, "D" : 500, "M" : 1000}

        integer = 0
        previous_char_val = mapping["M"]
        for c in s:
            c_int = mapping[c]
            integer += c_int
            if c_int > previous_char_val:
                integer -= (2 * previous_char_val)
        
            previous_char_val = c_int
            
        return integer
        